src/tron/tronEngine.js

```diff
@@ -0,0 +1,641 @@
+              if (valid) {
+                tronTx.blockNumber = jsonObj.blockNumber
+                tronTx.block_timestamp = jsonObj.blockTimeStamp
+              }
+
+              this.processTronApiTransaction(tronTx, currencyCode)
+            }
+            if (transactionsResults.data.length < NUM_TRANSACTIONS_TO_QUERY) {
+              break
+            }
+          } else {
+            checkAddressSuccess = false
+            this.log('checkTransactionsFetch inner loop invalid query results.')
+            break
+          }
+        }
+        start = end
+      }
+    } catch (e) {
+      this.log(
+        `Error checkTransactionsFetch ${currencyCode}: ${this.walletLocalData.publicKey}`,
+        e
+      )
+      checkAddressSuccess = false
+    }
+
+    if (checkAddressSuccess) {
+      this.tokenCheckTransactionsStatus[currencyCode] = 1
+      this.updateOnAddressesChecked()
+      return true
+    } else {
+      return false
+    }
+  }
+
+  async checkTransactionsInnerLoop() {
+    const blockHeight = Date.now()
+    let startTime: number = TIMESTAMP_BEFORE_TRX_LAUNCH
+    const promiseArray = []
+
+    if (
+      this.walletLocalData.lastAddressQueryHeight >
+      ADDRESS_QUERY_LOOKBACK_BLOCKS
+    ) {
+      // Only query for transactions as far back as ADDRESS_QUERY_LOOKBACK_BLOCKS from the last time we queried transactions
+      startTime =
+        this.walletLocalData.lastAddressQueryHeight -
+        ADDRESS_QUERY_LOOKBACK_BLOCKS
+    }
+
+    for (const currencyCode of this.walletLocalData.enabledTokens) {
+      promiseArray.push(this.checkTransactionsFetch(startTime, currencyCode))
+    }
+
+    let resultArray = []
+    try {
+      resultArray = await Promise.all(promiseArray)
+    } catch (e) {
+      this.log('Failed to query transactions')
+      this.log(e.name)
+      this.log(e.message)
+    }
+    let successCount = 0
+    for (const r of resultArray) {
+      if (r) successCount++
+    }
+    if (successCount === promiseArray.length) {
+      this.walletLocalData.lastAddressQueryHeight = blockHeight
+    }
+    if (this.transactionsChangedArray.length > 0) {
+      this.currencyEngineCallbacks.onTransactionsChanged(
+        this.transactionsChangedArray
+      )
+      this.transactionsChangedArray = []
+    }
+  }
+
+  async checkUpdateNetworkFees() {
+    try {
+      const jsonObj = await tronWeb.trx.getChainParameters()
+      const valid = validateObject(jsonObj, NetworkFeesSchema)
+
+      if (valid) {
+        // 1 SUN = 0.000001 TRX utilizing fromSun()
+        const networkFees: TronNetworkFees = {
+          createAccountFee: tronWeb.fromSun(
+            jsonObj.find(e => e.key === 'getCreateAccountFee').value
+          ),
+          transactionFee: tronWeb.fromSun(
+            jsonObj.find(e => e.key === 'getTransactionFee').value
+          )
+        }
+        // only update network fee if it's not the same
+        if (this.otherData.networkFees !== networkFees) {
```

Comment:
If will always be true. Either remove if statement or check actual values inside object

networkFees.createAccountFee !== this.otherDate.networkFees.createAccountFee
networkFees.transactionFee !== this.otherDate.networkFees.transactionFee